core/pkg/server/handler.go

```diff
@@ -939,6 +959,219 @@ func (h *Handler) handleTBrecord(record *service.Record) {
 	}
 }
 
+// handleHistory handles a history record. This is the main entry point for history records.
+// It is responsible for handling the history record internally, processing it,
+// and forwarding it to the Writer.
```

Comment:
IMO, repeating the method name in its documentation adds noise. Likewise, "handleHistory handles a history record" doesn't add much more than is present in the name. I'd structure this as

```
// Summary line.
//
// Description that can take multiple lines and so on and so forth
// and so on and so forth.
```

In this case, maybe something like

```suggestion
// The main entry point for history records.
//
// This processes a history record and forwards it to the writer.
```